modules/schema/src/main/resources/lucuma/odb/graphql/OdbSchema.graphql

```diff
@@ -9427,54 +9440,67 @@ uncharged.
 """
 interface TimeChargeDiscount {
 
-  "Time amount to discount from the program."
-  program: TimeSpan!
+  "The interval during which this discount applies."
+  interval: TimestampInterval!
 
   "Time amount to discount from the partner."
   partner: TimeSpan!
 
-  """
-  The atoms that are impacted (i.e., whose execution time is discounted) by this
-  discount.
-  """
-  atoms: [AtomRecord!]!
+  "Time amount to discount from the program."
+  program: TimeSpan!
 
-  "Optional additional detail."
-  comment: String
+  "Additional detail."
+  comment: String!
 
 }
 
-# TBD
-#type TimeChargeFaultDiscount implements TimeChargeDiscount
-#type TimeChargeOverlapDiscount implements TimeChargeDiscount
-#type TimeChargeWeatherDiscount implements TimeChargeDiscount
-
 """
-A time charge discount that is applied when one or more datasets are marked
-usable (as opposed to pas) or fail.
+A time charge discount that is applied when part of the observation is executed
+outside of nautical twilight boundaries.  The portion that occurs during day
+time is not charged.
 """
-type TimeChargeQaDiscount implements TimeChargeDiscount {
+type TimeChargeDaylightDiscount implements TimeChargeDiscount {
+
+  "The interval during which this discount applies."
+  interval: TimestampInterval!
+
+  "Time amount to discount from the partner."
+  partner: TimeSpan!
 
   "Time amount to discount from the program."
   program: TimeSpan!
 
+  "Additional detail."
+  comment: String!
+
+  "The site where the observation was executed."
+  site: Site!
+
+}
+
+type TimeChargeQaDiscount implements TimeChargeDiscount {
+
+  "The interval during which this discount applies."
+  interval: TimestampInterval!
+
   "Time amount to discount from the partner."
   partner: TimeSpan!
 
-  """
-  The atoms that are impacted (i.e., whose execution time is discounted) by this
-  discount.
-  """
-  atoms: [AtomRecord!]!
+  "Time amount to discount from the program."
+  program: TimeSpan!
 
-  "Optional additional detail."
-  comment: String
+  "Additional detail."
+  comment: String!
 
-  "Datasets with non-passing QA states."
+  "Datasets associated with the discount."
   datasets: [Dataset!]!
```

Comment:
I tried to make this into a query with offset and limit such as appears elsewhere but was soundly defeated by Grackle.  The only difference that I could appreciate from other cases (such as in the `VisitMapping`) was the usage of an intermediate link table between discounts and datasets.